### PR TITLE
Add ftl::fibtex

### LINKED ISSUE #64: Add cmpxchg to AtomicCounter to Enable Mutexes
In order to build a fiber based mutex AtomicCounter needs either cmpxchg. As for what to do with resuming the fibers, I think if the cmpxchg returns true, the fibers with the new value should be run.

### PR DESCRIPTION
**Pull request by RichieSams**
This is a re-submit of #66, after some fixes and rebasing/squashing.

Closes #64 

This adds a Fiber-aware mutex implementation. The member functions are named so they can be used with std::lockable-type classes (unique_lock, lock_guard, etc.)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,25 +30,31 @@ option(FTL_BUILD_TESTS "Build FiberTaskingLib tests" ON)
...
-# Define the guard page size
+# Add guard pages
 if (FTL_FIBER_STACK_GUARD_PAGES)
 	add_definitions(-DFTL_FIBER_STACK_GUARD_PAGES=1)
 endif()
 
-# CTest needs to be included as soon as possible
-if (FTL_BUILD_TESTS)
-	include(CTest)
+if (FTL_CPP_17)
```
Comment (martty) [inline CMakeLists.txt]: This is no longer the way to do this in cmake, use target_compile_features

**Comment by martty** [comment]
There are some bogus files checked in, could you remove?

**Comment by RichieSams** [comment]
Whicg bogus files? I only see the new fibtex files, formatting files, and the gitattributes file.
Screenshot_20200218-062328

**Comment by martty** [comment]
.ftl.dotSettings, plus run-clang-tidy.py seems like a file that belongs to clang tidy not us, but of this I am not sure. Are those linter/formatter files for the CI? Why are they in this PR?

File: CMakeLists.txt [hunk 2]
```diff
@@ -30,25 +30,31 @@ option(FTL_BUILD_TESTS "Build FiberTaskingLib tests" ON)
 option(FTL_BUILD_BENCHMARKS "Build FiberTaskingLib benchmarks" ON)
 option(FTL_VALGRIND "Link and test with Valgrind" OFF)
 option(FTL_FIBER_STACK_GUARD_PAGES "Add guard pages around the fiber stacks" OFF)
+option(FTL_CPP_17 "Enable C++17 features" OFF)
+option(FTL_WERROR "Promote complier warnings to errors." OFF)
```
Comment (martty) [inline CMakeLists.txt]: Typo

**Comment by RichieSams** [comment]
Ahhh, I understand. It was laziness. I just tacked those commits to this PR.
They're convenience scripts for running formatting / tidying. I'll move them to their own PR. (And we could potentially have CI run them and check for no changes)

**Comment by RichieSams** [comment]
Ok, pared down and fixed. The other commits will be in another PR after this is merged

**Pull request merged**